.travis.yml

```diff
@@ -3,23 +3,36 @@ language: python
 python:
 - 2.7
 
+addons:
+  firefox: latest
+  sauce_connect: true
+  apt:
+    sources:
+    - google-chrome
+
 branches:
   only:
   - master
   - develop
 
 env:
-- RUN_BACKEND_TESTS=false RUN_FRONTEND_TESTS=false RUN_LINT=true REPORT_BACKEND_COVERAGE=false
-- RUN_BACKEND_TESTS=false RUN_FRONTEND_TESTS=true RUN_LINT=false REPORT_BACKEND_COVERAGE=false
-- RUN_BACKEND_TESTS=true RUN_FRONTEND_TESTS=false RUN_LINT=false REPORT_BACKEND_COVERAGE=false
-- RUN_BACKEND_TESTS=true RUN_FRONTEND_TESTS=false RUN_LINT=false REPORT_BACKEND_COVERAGE=true
-
-
+  matrix:
+    - RUN_E2E_TESTS=true
+    - RUN_BACKEND_TESTS=false RUN_FRONTEND_TESTS=false RUN_LINT=true REPORT_BACKEND_COVERAGE=false
+    - RUN_BACKEND_TESTS=false RUN_FRONTEND_TESTS=true RUN_LINT=false REPORT_BACKEND_COVERAGE=false
+    - RUN_BACKEND_TESTS=true RUN_FRONTEND_TESTS=false RUN_LINT=false REPORT_BACKEND_COVERAGE=false
+    - RUN_BACKEND_TESTS=true RUN_FRONTEND_TESTS=false RUN_LINT=false REPORT_BACKEND_COVERAGE=true
+  # Connect to SauceLabs by adding encripted username and Sauce Key
+  # They will be available globaly in travis env to be used by core/tests/protractor.conf.js
+  global:
+    - secure: J2nmHZz/cF0N5QdhbT7VCgXTJWXHlLnr4Eo1DtaVOAs5IUT/FjzZE/ztBj5K9cTjDVu31fxASibpTL70h3QqrWNmPpf0+3Ea4TY4NI3KCRCNvNgCy6pjgZr6cK4gSHuqdApRLp0n1RiXlJn5mcHTpQR0vIHyJPfr/+SSxJ9ygNY6OgzrvG+qMdIrpjcVJ12mMLnT7o5OCyraY5Gazg92mVxRBOKKVJBG52vvtpfulPuSpv6Ptwi+QV9JOaRBCvWi2SQqWuayKjDN75cniSH9UiANlRo+Ng2TEm+HeK+u8kEPTR/Mt8JcGz6LIQoYzsT4WpTtdAvvwexTj7i48mwoWqJ7RbzljSiRw31mjL7NL5ir5PzK/gehXqOD9ClYp1FLrJOKvVLgIL+HufnR08qLdWk22qJN9o/GRZkXROAG5MFXXPGsuyCyhMjdwJI4b/0NIfCHhb8hd48t4lvjcdI2OgOM53SwjdWEZ03fm7RQcIp1iuH2bO1UZ1epF+06JETEu4NWiAX1s5o1PudgKOIDyAkTn6+iJ4oxxa0dd++j9XNk906HZPZxFfj4qpahUX4zVSCrEg8No+uxICmYI+7uiITB11Kg8ex0EoYtydRyif2u8j9nbbvi58QA4UOP77GAy/mLuYN6zXmKRb8OrG/u9Np4JnFsmhj4fxjmEbPAV3Y=
+    - secure: l0ZPEIAWeVtc8sITt31908AFCaJB+yb/rP8QpXUzYZZqMU00J/jA1Vn3P925+tyD2TnSks80+I/8s2quq99O9Ke4lziLh6syu0yaSFB0Q29EiN/1sKZ6UN//0qJr1C44hw3bHiR8bfJBRcx5CxnW7KuGZEbatM0M0Yyo/wRT2tMh2PzoqbCvgv+XuFCXWVmjgz87SyEtfFyoLmT1iRTjj90GNbECy/X4hu4Pdhjd0V96gpbrtmlyt1WR3H4wlYLuYFmt9Zh5XPANXuxn0RPX2Y5U7caqNBs4k+8Xl/ubd+v/oTsuOS0N+QE/QYiu+yY7jmKYHwhq+KVYO8AMKD8y44U3cwmaFckj4OoQzgX32B01rRQGHWPNyz7RWzfdNbjcgYShCUGzNpeGQ7C3WvY9MFtSgAfgUTlRqzaVRmgH+vLtVSrfpl6pH2CeiCrfUh7ZgoW48xIUsbiBhc6us1YyHlH8vVC9UAIxO8hlNWCvlHQkvXc77HR2Gzol7zdsJWqBlU39eG5hPcfyDMfnEzqSySggoJ8bIBi+uaTYcxiKTW91iL3hjnPTgGrHQhQJ9aCzqypmSW5+CJv89wZG8Leme6wkpZ56G5/m49hN1nwr6KTRl8dV+uKwD5BRTXHypenYwU3/avI5LW7SLyOo1Oo+eZXoZo23YkHKhuLV5VsNc0M=
 matrix:
   allow_failures:
   # The backend tests, with coverage, take too long to run, so we make this
   # optional.
   - env: RUN_BACKEND_TESTS=true RUN_FRONTEND_TESTS=false RUN_LINT=false REPORT_BACKEND_COVERAGE=true
+  - env: RUN_E2E_TESTS=true
```

Comment:
Add comment: The e2e tests are not succeeding consistently, and fail for PRs from outside the Oppia repository, so we're making them optional for now. However, we plan to make them mandatory once both these issues are addressed.